packages/api/__tests__/utility.ts

```diff
@@ -0,0 +1,129 @@
+import {MongoDBAdapter} from '@wepublish/api-db-mongodb'
+import {URL} from 'url'
+import {
+  Author,
+  contextFromRequest,
+  GraphQLWepublishPublicSchema,
+  GraphQLWepublishSchema,
+  OptionalUserSession,
+  PublicArticle,
+  PublicPage,
+  URLAdapter
+} from '../src'
+import {ApolloServer} from 'apollo-server'
+import {createTestClient} from 'apollo-server-testing'
+import {ApolloServerTestClient} from 'apollo-server-testing/dist/createTestClient'
+import {KarmaMediaAdapter} from '@wepublish/api-media-karma/src'
+
+export interface TestClient {
+  dbAdapter: MongoDBAdapter
+  testClientPublic: ApolloServerTestClient
+  testClientPrivate: ApolloServerTestClient
+}
+
+class ExampleURLAdapter implements URLAdapter {
+  getPublicArticleURL(article: PublicArticle): string {
+    return `https://demo.wepublish.ch/article/${article.id}/${article.slug}`
+  }
+
+  getPublicPageURL(page: PublicPage): string {
+    return `https://demo.wepublish.ch/page/${page.id}/${page.slug}`
+  }
+
+  getAuthorURL(author: Author): string {
+    return `https://demo.wepublish.ch/author/${author.slug || author.id}`
+  }
+}
+
+export async function createGraphQLTestClientWithMongoDB(): Promise<TestClient> {
+  if (!process.env.TEST_MONGO_URL) {
+    throw new Error('TEST_MONGO_URL not defined')
+  }
+  let adminUser
+
+  await MongoDBAdapter.initialize({
+    url: process.env.TEST_MONGO_URL!,
```

Comment:
what does the `!` at the end of `url: process.env.TEST_MONGO_URL!`

Reply:
https://stackoverflow.com/questions/42273853/in-typescript-what-is-the-exclamation-mark-bang-operator-when-dereferenci

Reply:
thanks